File: Backend/app/models/post.py
"""
Event post / feed model: registered users can post on events.
"""
from datetime import datetime
from sqlalchemy import Text, Integer, DateTime, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class EventPost(Base):
    __tablename__ = "event_posts"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    event_id: Mapped[int] = mapped_column(ForeignKey("events.id"), nullable=False, index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False, index=True)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=datetime.utcnow)

    event = relationship("Event", back_populates="posts")
    user = relationship("User", back_populates="posts")

File: FrontEnd/lib/models/post.dart
class EventPost {
  final int id;
  final int eventId;
  final int userId;
  final String? authorName;
  final String content;
  final DateTime createdAt;

  EventPost({
    required this.id,
    required this.eventId,
    required this.userId,
    this.authorName,
    required this.content,
    required this.createdAt,
  });

  factory EventPost.fromJson(Map<String, dynamic> json) {
    return EventPost(
      id: (json['id'] as num?)?.toInt() ?? 0,
      eventId: (json['event_id'] as num?)?.toInt() ?? 0,
      userId: (json['user_id'] as num?)?.toInt() ?? 0,
      authorName: json['author_name'] as String?,
      content: json['content'] as String? ?? '',
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}
